### ISSUE TITLE
general discussion

### ISSUE THREAD

**Issue by krakrukra** [krakrukra/PocketAdmin#1]
this is created just in case you have some very small question or comment and you do not feel like creating a new github issue for it

**Comment by SamPatt** [comment]
I noticed you using an OpenBazaar link. You can use a .com link to better share online.

https://openbazaar.com/store/QmeCrxkz8J1pvBx4nVE7EgZNkLfMftmKtz3dc5oo4bPgqr/pocketadmin-an-open-source-keystroke-injection-device-badusb

**Comment by krakrukra** [comment]
thanks, I added the extra link

**Comment by BlueskyFR** [comment]
Hey!
Can I build it myself?
If yes, what parts should I use and what is the flashing process?
Thanks in advance!

**Comment by krakrukra** [comment]
Yes, you can. The parts are all listed in /hardware/BOM.txt (though, check the kicad files too)
and the flashing process (at least on a debian system) consists of installing openocd, make
and typing this in the terminal: make upload

but I probably will make a video about it on my youtube channel, to make things clear.

**Comment by krakrukra** [comment]
OK, so here is a video on how to build it:
https://www.youtube.com/watch?v=cfud5Dq_w2M

**Comment by kivik92** [comment]
Добрый день а как обстоят дела из keyboard_layout?
Тоесть если зарание неивзестно какая раскладка клавиатуры установлена?
То что с этим делать?

**Comment by krakrukra** [comment]
если заранее не известно какая раскладка выбрана,  то нужную раскладку
надо выбрать каким-то способом. Например такой способ для windows 10
я показывал тут https://www.youtube.com/watch?v=o4rd-4753e0
смотреть нужно с 11:55 по 13:05

**Comment by kivik92** [comment]
> если заранее не известно какая раскладка выбрана, то нужную раскладку
> надо выбрать каким-то способом. Например такой способ для windows 10
> я показывал тут https://www.youtube.com/watch?v=o4rd-4753e0
> смотреть нужно с 11:55 по 13:05

Ок. А если 3  языка установлено на ПК? То как быть в этом случае?

**Comment by krakrukra** [comment]
ну клавиша END выберет самый нижний в списке. если не жать сразу ENTER,
а сначала нажать UPARROW, то будет выбран второй снизу итд.

**Comment by krakrukra** [comment]
а хотя нет, только что проверил. нужно не стрелками выбирать а ещё раз 
GUI SPACE  нажать. То есть через END выбрать нижний в списке, и с этого момента
нажать GUI SPACE 1 раз чтобы выбрать первый в списке, 2 раза чтобы выбрать 
второй, итд.

**Comment by yeboiiii** [comment]
Is it possible to use it to just open a cmd window with admin privileges? (without the privileges yourself)

**Comment by krakrukra** [comment]
It might be possible, but I personally do not know how. It will probably be 
something like silentUACbypass payload, where you take advantage of
task sceduler or some other thing. But this payload does require you to have
admin rights on the system.

**Comment by TinSnail** [comment]
Hello, I have tried to alter the length of the serial number (using a v1.2 device) from:

[usb_rodata.h]
unsigned short StringDescriptor_1[13] = 

to 

 unsigned short StringDescriptor_1[21] =

The serial number length is still 13 characters when reviewed using ChipGenius

I then changed 

[main.c]
extern unsigned short StringDescriptor_1[13];

to
extern unsigned short StringDescriptor_1[21];

The serial number still remained 13 characters. 

It appears that there is a keyword to change the serial length:
checkKeyword("SERIAL ")

and looking into the config the serial is limited to 12 characters (v1.3 wiki):

https://github.com/krakrukra/PocketAdmin/wiki/configuration

Can the serial number be longer than 12 characters? If so what should be changed?

Thank you.

**Comment by krakrukra** [comment]
Yes, it can be longer. You should also change the first element in the a StringDescriptor_1 array, from
0x031A to 0x032A (because LSB there actually specifies length of the descriptor in bytes, 42 in your case)

"SERIAL" pre-configuration command is actually changing the value of the serial number, not it's length.
This is so people can avoid flashing different firmware in case they want to change the serial number.
In your case, you should find a function in main.c called setSerialNumber() and replace 2 instances of
"i<12" to "i<20" and this command then should work properly for 20-character long serial number.

**Comment by TinSnail** [comment]
Thank you. I will try this.

**Comment by krakrukra** [comment]
Actually wait, I missed something. Are you trying to flash rev. 1.2 device with current firmware?
This will not work

**Comment by TinSnail** [comment]
No it is old firmware. I can see the new firmware has the SERIAL feature. I have not moved to the new firmware as I need to read more about the differences. My apologies I have the modified old firmware on Debian an the new firmware on a windows machine. It took a while to find the serial string with checks using ChipGenius.

**Comment by krakrukra** [comment]
Actually, here is the repository with firmware for rev 1.2 devices (inside /firmware_12005/):
https://github.com/krakrukra/12004-pocketadmin-wiki
This is the latest one for 1.2 devices that I have released. "SERIAL" command should actually be there.

**Comment by TinSnail** [comment]
Thank you. I will try this method as the other embedded values, if required, can also be changed easily using the keyword method.

**Comment by TinSnail** [comment]
WIth StringDescriptor_1 array, updated from 0x031A to 0x032A the new serial worked. Thank you.

**Comment by yeboiiii** [comment]
Is it possible to combine a rubber ducky usb, a kill usb and a storage device (64 or 128gb) into one? (you would toggle it with a small switch that has 3 positions)

**Comment by krakrukra** [comment]
It is possible, but it will not be practical.
